### DISCUSSION TITLE
How to disable Zoom By Selection (Double click and drag/right click and drag zoom)?

### DISCUSSION THREAD

**Discussion by Thutek** [Live-Charts/LiveCharts2#1408] [Q&A] [answered]
I would like to disable Zoom By Selection in my application. By that I mean that I want the user to be able to use the mouse wheel or pinch to zoom for zooming, but I don't want them to double click and drag to zoom. My touch monitor sometimes registers to fingers close together as a single finger that's tapping on the screen which randomly triggers the zoom by selection behavior when I don't want it to.

 I found documentation on this feature from this page: https://livecharts.dev/docs/Avalonia/2.0.0-rc2/samples.lines.zoom#zoom-by-selection which then links to the following page saying that you can customize the trigger of the function here: https://livecharts.dev/docs/avalonia/2.0.0-rc2/CartesianChart.Axes%20properties#zooming-and-panning. When I search for 'Zoom by Selection' I can't seem to find any information on how to disable it or capture the trigger to potentially ignore it. 

I've also looked in the API browser and searched for 'Zoom By Selection', 'Selection', and 'Zoom' and didn't find anything explaining how to disable this feature. Does anyone have some documentation they could point me towards on how to disable this function?

**Comment by Thutek** [comment]
Update: I've tracked down the method that controls the zoom by selection logic (I think). It's in /src/skiasharp/LiveChartsCore.SkiaSharp.Avalonia/CartesianChart.axaml.cs  and the code is:

```
private void CartesianChart_PointerPressed(object? sender, PointerPressedEventArgs e)
    {
        if (e.KeyModifiers > 0) return;
        var p = e.GetPosition(this);

        if (PointerPressedCommand is not null)
        {
            var args = new PointerCommandArgs(this, new(p.X, p.Y), e);
            if (PointerPressedCommand.CanExecute(args)) PointerPressedCommand.Execute(args);
        }

        var isSecondary =
            e.GetCurrentPoint(this).Properties.IsRightButtonPressed ||
            (DateTime.Now - _lastPresed).TotalMilliseconds < 500;

        _core?.InvokePointerDown(new LvcPoint((float)p.X, (float)p.Y), isSecondary);
        _lastPresed = DateTime.Now;
    }

```

IsSecondary looks like the trigger that I'm looking for.

I think I could potentially avoid IsSecondary from being set to true by overriding the method and not calling this method on RightButton presses or double taps within 500ms, but I'm hoping there's a more straightforward way to disable this functionality that I'm not noticing.

InvokePointerDown for the cartesian chart (what I'm currently using) doesn't appear to have any method to ignore the secondary option either. (src/LiveChartsCore/CartesianChart.cs)
)

**Reply by rubiomiguel06** [reply]
Did you find any way to disable zoom by selection? I get the same inconsistent behavior when panning and zooming on touch devices

**Comment by beto-rodriguez** [comment, marked as answer]
This is now supported https://livecharts.dev/docs/Maui/2.0.0-rc6/samples.lines.zoom 

```xml
<lvc:CartesianChart ZoomMode="X,NoZoomBySection" />
```

In general, there are multiple improvements in zooming/panning, and the article in the web explains how to build custom features.
